http3/response_writer.go

```diff
@@ -200,7 +204,29 @@ func (w *responseWriter) writeHeader(status int) error {
 		return err
 	}
 
+	// Handle trailer fields
+	if vals, ok := w.header["Trailer"]; ok {
+		trailers := make([]string, 0, len(vals))
+		for _, val := range vals {
+			for _, trailer := range strings.Split(val, ",") {
```

Comment:
I didn't realize you can comma-separate the trailer names here, thanks for the example of that @marten-seemann. This code is a bit uglier but it now handles this better.

Reply:
@sudorandom Should we revert the `strings.Split` change introduced in cffa7c41ba2258d2a00a16fa8845dd06c566ac8b?

Reply:
I think this part is still required so the server can know which headers are actually trailers to exclude them in the header frame and include them into the trailers frame. I already changed this code to avoid modifying the "Trailer" header but I do think this code is required to handle [this example](https://pkg.go.dev/net/http#example-ResponseWriter-Trailers) from the server side.

I looked a bit at the net/http implementation, and it also has to do something similar with splitting values from the Trailer header before sending the headers on the server side:
- https://github.com/golang/go/blob/master/src/net/http/response.go#L297
- https://github.com/golang/go/blob/master/src/net/http/transfer.go#L544
- https://github.com/golang/go/blob/master/src/net/http/transfer.go#L786

As a note, `foreachHeaderElement` actually handles comma-separated header values.

Reply:
You're right, we need to split the list to avoid sending out the value as a header.

Do we need to make any changes here to resolve https://github.com/quic-go/quic-go/pull/4656#discussion_r1743240176?

Reply:
No, I can resolve it in #4656